starter-code/src/viking.js

```diff
@@ -1,11 +1,105 @@
 // Soldier
-function Soldier() {}
+
+function Soldier(health, strength) {
+    this.health = health;
+    this.strength = strength;
+}
+
+
+Soldier.prototype.attack = function () {
+    return this.strength;
+}
+
+Soldier.prototype.receiveDamage = function (damage) {
+    this.health = this.health - damage;
+}
+
 
 // Viking
-function Viking() {}
+function Viking(name, health, strength) {
+    Soldier.call(this, health,strength);
+    this.name = name;
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.constructor = Viking;
+
+Viking.prototype.receiveDamage = function (damage) {
+    this.health = this.health - damage;
+    if (this.health > 0) {
+        return this.name + ' has received '+ damage + ' points of damage'
+    }else {
+        return this.name + ' has died in act of combat'
+    }
+}
+
+Viking.prototype.battleCry = function(){
+    return 'Odin Owns You All!'
+}
 
 // Saxon
-function Saxon() {}
+function Saxon(health, strength) {
+    Soldier.call(this, health, strength);
+}
+Saxon.prototype = Object.create(Soldier.prototype);
+Saxon.prototype.constructor = Saxon;
+
+Saxon.prototype.receiveDamage = function (damage) {
+    this.health = this.health - damage;
+    if (this.health > 0) {
+        return 'A Saxon has received ' +damage + ' points of damage' 
+    }else {
+        return 'A Saxon has died in combat'
+    }
+}
 
 // War
-function War() {}
+function War() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+}
+
+War.prototype.addViking = function(Viking){
+    this.vikingArmy.push(Viking) 
+}
+
+War.prototype.addSaxon = function(Saxon){
+    this.saxonArmy.push(Saxon) 
+}
+
+War.prototype.vikingAttack = function () {
+    var randomViking = Math.floor(Math.random() * this.vikingArmy.length);
+    var randomSaxon = Math.floor(Math.random() * this.saxonArmy.length);
+
+    var warResult = this.saxonArmy[randomSaxon].receiveDamage(this.vikingArmy[randomViking].strength);
```

Comment:
Don't access the viking strength directly, you have a method for that: attack().